tensorflow/compiler/xla/service/heap_simulator.cc

```diff
@@ -409,11 +409,15 @@ HeapSimulator::Result<HloValue> HeapSimulator::Finish() {
   // Post-process the result to add chunks for shared buffers.  An empty chunk
   // map means that either no buffers were allocated, or the heap was only
   // collecting statistics, e.g. NoFragmentationStatsHeap.
-  if (!result.chunk_map.empty()) {
+  size_t total_chunk_count = 0;
+  absl::c_for_each(result.heap_results, [&](const HeapResult<HloValue>& hr) {
```

Comment:
`absl::c_accumulate` seems to fit better

Reply:
Shouldn't `absl::c_accumulate` require the type of the container to be implicitly convertible to the accumulated type? It is not a valid condition here.

Reply:
Not when you provide a custom reducer.

Reply:
`Ret fun(const Type1 &a, const Type2 &b);`
The type `Type1` must be such that an object of type `T` can be implicitly converted to Type1. ...
See https://en.cppreference.com/w/cpp/algorithm/accumulate

Or am I missing something?

Reply:
I think it is possible with C++17 under a new function name, but not C++14 to my knowledge.
Does the public TF started to use C++17?

Reply:
@nouiz TF is still very much C++14, but this is absl --- one of the reasons of its existance is the ability to use newer C++ features while using an older dialect.

Reply:
> The type Type1 must be such that an object of type T can be implicitly converted to Type1. ...

But not `Type2`, right?

Reply:
Ah, I missed that. Revised. Thanks for the pointer!